_posts/2019-08-23-unet/unet.Rmd

```diff
@@ -0,0 +1,437 @@
+---
+title: "Image segmentation with U-Net"
+description: > 
+ In image segmentation, every pixel of an image is assigned a class. Depending on the application, classes could be different cell types; or the task could be binary, as in "cancer cell yes or no?". Area of application nonwithstanding, the established neural network architecture of choice is U-Net. In this post, we show how to preprocess data and train a U-Net model on the Kaggle Carvana image segmentation data.
+ 
+author:
+  - name: Daniel Falbel and Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: falbelkeydana2019unet
+date: 08-23-2019
+categories:
+  - Images
+  - Introductions
+  - Keras
+bibliography: bibliography.bib
+output:
+  distill::distill_article:
+    self_contained: false
+preview: images/unet.png
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+Sure, it is nice when I have a picture of some object, and a neural network can tell me what kind of object that is. More realistically, there might be several salient objects in that picture, and it tells me what they are, and where they are. The latter task especially (known as _object detection_) seems prototypical of contemporary AI applications that at the same time are fascinating intellectually and questionable ethically. It's different with the subject of this post: Successful _image segmentation_ has a lot of undeniably useful applications. For example, it is a sine qua non in medicine, neuroscience, biology and other life sciences.
+
+So what, technically, is image segmentation, and how can we train a neural network to do it?
+
+## Image segmentation in a nutshell
+
+Say we have an image with a bunch of cats in it. In _classification_, the question is "what's that?", and the answer we want to hear is: "cat". In _object detection_, we again ask "what's that", but now that "what" is implicitly plural, and we expect an answer like "there's a cat, a cat and a cat, and they're here, here, and here" (imagine the network pointing, by means of drawing _bounding boxes_, i.e., rectangles around the detected objects). In _segmentation_, we want more: We want the whole image covered by "boxes" -- which aren't boxes anymore, but unions of pixel-size "boxlets" -- or put differently: __We want the network to label every single pixel in the image.__
```

Comment:
```suggestion
Say we have an image with a bunch of cats in it. In _classification_, the question is "what's that?", and the answer we want to hear is: "cat". In _object detection_, we again ask "what's that", but now that "what" is implicitly plural, and we expect an answer like "there's a cat, a cat, and a cat, and they're here, here, and here" (imagine the network pointing, by means of drawing _bounding boxes_, i.e., rectangles around the detected objects). In _segmentation_, we want more: We want the whole image covered by "boxes" -- which aren't boxes anymore, but unions of pixel-size "boxlets" -- or put differently: __We want the network to label every single pixel in the image.__
```

Reply:
Added a comma so the "cat, cat, and a cat" has the same structure as "here, here, and here".

Reply:
lol, sure, how could I ;-))